jupyterhub/handlers/base.py

```diff
@@ -654,8 +654,42 @@ def get_next_url(self, user=None, default=None):
                     next_url = url_path_join(self.hub.base_url, 'spawn')
             else:
                 next_url = url_path_join(self.hub.base_url, 'home')
+
+        next_url = self.append_query_parameters(next_url, exclude=['next'])
```

Comment:
I think this addition of params is the one that's definitely correct, adding params where `next` is already defined is probably not correct.